tests/test_gp_var_dist.py

```diff
@@ -39,149 +39,62 @@ def test_toeplitz_GP_lat_prior(use_fast_toeplitz=True):
-    n = 50  # number of neurons
-    m = 80  # number of conditions / time points
-    n_z = 15  # number of inducing points
-    n_samples = 10  # number of samples
-    Poisson = False
-
-    #generate from GPFA generative model
-    ts = np.array([np.arange(m) for nsamp in range(n_samples)])[:, None, :]
-
-    dts_sq = (ts[..., None] - ts[..., None, :])**2  #(n_samples x 1 x m x m)
-    dts_sq = np.sum(dts_sq, axis=-3)  #(n_samples x m x m)
-    K = np.exp(-dts_sq / (2 * 7**2)) + 1e-6 * np.eye(m)[None, ...]
-    L = np.linalg.cholesky(K)
-    us = np.random.normal(0, 1, size=(m, d))
-    xs = L @ us  #(n_samples x m x d)
-    print('xs:', xs.shape)
-    w = np.random.normal(0, 1, size=(n, d))
-    Y = w @ xs.transpose(0, 2, 1)  #(n_samples x n x m)
-    if Poisson:
-        Y = np.random.poisson(2 * (Y - np.amin(Y)))
-    else:
-        Y = Y + np.random.normal(0, 0.2, size=Y.shape)
-    print('Y:', Y.shape, np.std(Y), np.quantile(Y, 0.99))
-
-    data = torch.tensor(Y, device=device, dtype=torch.get_default_dtype())
-    # specify manifold, kernel and rdist
-    manif = mgp.manifolds.Euclid(m, dfit)
-    #kernel = mgp.kernels.Linear(n, dfit)
-    torch.manual_seed(0)
-    lat_dist = mgp.rdist.EP_GP(manif,
-                               m,
-                               n_samples,
-                               torch.Tensor(ts),
-                               Y=Y,
-                               initialization='fa',
-                               use_fast_toeplitz=False)
-    # break symmetry in K_half@S
-    _scale = torch.ones(n_samples, lat_dist.d, m) * .2 * (
-        1 + torch.randn(m) / 100)  #n_diag x T
-    lat_dist._scale = nn.Parameter(data=inv_softplus(_scale),
-                                   requires_grad=True)
-    sample1, sample2 = lat_dist.sample((2,))
-    prms1, prms2 = lat_dist.prms
-
-    torch.manual_seed(0)
-    lat_dist_toeplitz = mgp.rdist.EP_GP(manif,
-                                        m,
-                                        n_samples,
-                                        torch.Tensor(ts),
-                                        Y=Y,
-                                        initialization='fa',
-                                        use_fast_toeplitz=True)
-    _scale = torch.ones(n_samples, d, m) * .2 * (1 + torch.randn(m) / 100
-                                                )  #n_diag x T
-    lat_dist_toeplitz._scale = nn.Parameter(data=inv_softplus(_scale),
-                                            requires_grad=True)
-
-    sample_toeplitz1, sample_toeplitz2 = lat_dist_toeplitz.sample((2,))
-    prms_toeplitz1, prms_toeplitz2 = lat_dist_toeplitz.prms
-
-    assert torch.allclose(sample1, sample_toeplitz1)
-    assert torch.allclose(prms1, prms_toeplitz1)
-    assert torch.allclose(sample2, sample_toeplitz2)
-    assert torch.allclose(prms2[..., 0], prms_toeplitz2)
-    assert torch.allclose(prms2[..., 0, :], prms_toeplitz2)
-
-
 if __name__ == '__main__':
```

Comment:
It'd be good to add a test on the K^1/2 K^1/2 being approximately equal to K, at least in the middle (I know we verified today, but it would be easy and nice to ahve an automated check of that)

Reply:
I've included a test for this now